settings.gradle.kts

```diff
@@ -56,12 +56,14 @@ pluginManagement {
   }
 }
 
-@Suppress("UnstableApiUsage") dependencyResolutionManagement { repositories { mavenCentral() } }
+plugins { id("org.gradle.toolchains.foojay-resolver-convention") version ("0.8.0") }
 
-val javaVersion = JavaVersion.current()
+@Suppress("UnstableApiUsage") dependencyResolutionManagement { repositories { mavenCentral() } }
 
-require(javaVersion.isCompatibleWith(JavaVersion.VERSION_17)) {
-  "Project requires Java 17 or higher, but found ${javaVersion.majorVersion}."
+JavaVersion.current().let { toolchain ->
+  require(toolchain.isCompatibleWith(JavaVersion.VERSION_21)) {
+    "Project requires Java 21+ to build, but found ${toolchain.majorVersion}."
+  }
```

Comment:
This block is no longer necessary (your system Java has no bearing on the Java used for compiling/running)